src/helpers.ts

```diff
@@ -6,7 +6,7 @@ import {
   NetworkConfig,
 } from "hardhat/types";
 
-import type { SignerWithAddress } from "./signer-with-address";
+import type { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
```

Comment:
we cannot import anything from `@nomiclabs/hardhat-ethers`

We have to reimplement it. This is because currently hardhat-deploy-ethers cannot be an extension of hardhat-ethers and thus have to be installed as an alias to `@nomiclabs/hardhat-ethers` so that plugin that have harcoded dependencies on `@nomiclabs/hardhat-ethers` continue to work. This means hardhat-ethers and hardhat-deploy-ethers cannot coexists

Reply:
We could `import "@alias/hardhat-ethers"` and add `"@alias/hardhat-ethers": "npm:@nomiclabs/hardhat-ethers"` to peerDependencies?

Reply:
I would rather avoid this. the code is already a fork of hardhat-ethers and using the code as is should be relatively easy, unless a lot changes since last time I checked

Reply:
There have been quite a bit of changes in upstream to the directory structure and project layout.

We could add @nomiclabs/hardhat-ethers  as a runtime dependency to this package? That remove the need to inform users of a  new peer dep?

Reply:
As I said, this is not possible due to the way hardhat plugin are imported and that some plugin have hardcoded dependencies on @nomiclabs/hardhat-ethers

in the future the proper way is to make hardhat-deploy-ethers an extension of hardhat-ethers (and so implement only extra functions) but this is not possible right now, mainly due to https://github.com/nomiclabs/hardhat/issues/1040

Reply:
This current branch is fully working for me, so not sure what is the issue.

I suggest testing this branch out for yourself before making a hasty conclusion.

Reply:
Did you try with plugin like hardhat-waffle that have a hardcoded dependencies on hardhat-ethers ?
Do you have a repo where you test that branch ?

Reply:
If that works, I am  fine, but this need to be tested with all the scenario that caused issue in the past before I can merge

Reply:
Also I am not a fan of the extra peer dependency, especially as alias

why is that necessary ?